common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1197,15 +1230,15 @@ public void onData(String streamId, ByteBuffer buf) throws IOException {
             appShuffleInfo.shuffles.get(partitionInfo.appAttemptShuffleMergeId.shuffleId);
         if (isStale(info, partitionInfo.appAttemptShuffleMergeId.shuffleMergeId) ||
```

Comment:
QQ: Do we want to capture stale pushes ?

Reply:
Any thoughts ? Thx

Reply:
Thanks for the suggestion. It seems that a stale block push can appear during indeterminate stage retries ([reference](https://github.com/apache/spark/blob/05cd5f97c3dea25dacdbdb319243cdab9667c774/common/network-common/src/main/java/org/apache/spark/network/server/BlockPushNonFatalFailure.java#L132)). Not sure how necessary capturing this would be, it can be captured similar to capturing `lateBlockPushes` @mridulm .

Reply:
I would look at these metrics as a means for identifying the effectiveness of push based shuffle, what is impacting the inefficiencies (how many stale blocks, how many late blocks, how that distribution is changing with time), etc.
These will correlate in some way to overall impact on when push based shuffle was used, when normal shuffle was used, and what the network and task impact of those are.

Do let me know if you have thoughts !

Reply:
Make sense. Added `staleBlockPushes`, PTAL.
WIP to add UT.